component/loki.libsonnet

```diff
@@ -138,6 +138,190 @@ local operator_metrics_sa_token =
+    metadata: {
+      labels: {
+        name: 'loki-ingester-check',
+      },
+      name: 'loki-ingester-check',
+      namespace: 'openshift-logging',
+    },
+  },
+  {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'Role',
+    metadata: {
+      labels: {
+        name: 'loki-ingester-check',
+      },
+      name: 'loki-ingester-check',
+      namespace: 'openshift-logging',
+    },
+    rules: [ {
+      apiGroups: [ '' ],
+      resources: [ 'pods', 'pods/exec' ],
+      verbs: [
+        'get',
+        'list',
+        'watch',
+        'create',
+        'delete',
+        'patch',
+        'update',
+      ],
+    } ],
+  },
+  {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'RoleBinding',
+    metadata: {
+      labels: {
+        name: 'loki-ingester-check',
+      },
+      name: 'loki-ingester-check',
+      namespace: 'openshift-logging',
+    },
+    roleRef: {
+      apiGroup: 'rbac.authorization.k8s.io',
+      kind: 'Role',
+      name: 'loki-ingester-check',
+    },
+    subjects: [ {
+      kind: 'ServiceAccount',
+      name: 'loki-ingester-check',
+    } ],
+  },
+  {
+    apiVersion: 'v1',
+    kind: 'ConfigMap',
+    metadata: {
+      labels: {
+        name: 'loki-ingester-check',
+      },
+      name: 'loki-ingester-check',
+      namespace: 'openshift-logging',
+    },
+    data: {
+      'wal-check.sh': |||
+        #!/bin/bash
+
+        function check_pod() {
+          POD_NAME="loki-ingester-${1}"
+          echo "checking POD ${POD_NAME}"
+          PHASE=$(kubectl -n openshift-logging get po ${POD_NAME} -oyaml | yq '.status.phase')
+          if [ ${PHASE} != "Running" ]; then
+            return 0
+          fi
+
+          READY=$(kubectl -n openshift-logging get po ${POD_NAME} -oyaml | yq '.status.conditions[] | select(.type == "ContainersReady") | .status')
+          if [ ${READY} == "True" ]; then
+            return 0
+          fi
+
+          return 1
+        }
+
+        function check_dir() {
+          shopt -s extglob
+          POD_NAME="loki-ingester-${1}"
+          echo "checking DIR ${POD_NAME}"
+          DIR_CHP=$(kubectl -n openshift-logging exec -i ${POD_NAME} -- ls /tmp/wal | grep -o "^checkpoint\.[0-9]*$")
+          PATTERN=$(echo ${DIR_CHP} | sed 's/[^0-9]*//g')
+          DIR_WAL=$(kubectl -n openshift-logging exec -i ${POD_NAME} -- ls /tmp/wal | grep -o "^0*${PATTERN}$")
+          if [ -z $DIR_WAL ]; then
+            kubectl -n openshift-logging exec -i ${POD_NAME} -- rm -rf /tmp/wal/${DIR_CHP}
+            kubectl -n openshift-logging delete po ${POD_NAME}
+          fi
+        }
+
+        function fix_pod() {
+          check_pod $1
+          STATE=$?
+          if [ ${STATE} -gt 0 ]; then
+            echo "stuck POD, waiting ${SLEEP_TIME}"
+            sleep ${SLEEP_TIME}
+            check_pod $1
+            STATE=$?
+          fi
+
+          if [ ${STATE} -gt 0 ]; then
+            check_dir $1
+            exit 0
```

Comment:
This confused me for a bit, might make sense to add a comment why it's correct to immediately exit after we fix a pod.